From 8e51bd4121fcea021acc3c0e9da966e7b7e01e9b Mon Sep 17 00:00:00 2001
From: Alex Ermolov <aaermolov@gmail.com>
Date: Thu, 11 Apr 2019 23:40:55 +0300
Subject: [PATCH] xkb: forcefully reset keyboard layout group to 0

---
 i3lock.c | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/i3lock.c b/i3lock.c
index e855ad9..50cad31 100644
--- a/i3lock.c
+++ b/i3lock.c
@@ -1868,8 +1868,6 @@ int main(int argc, char *argv[]) {
         xcb_connection_has_error(conn))
             errx(EXIT_FAILURE, "Could not connect to X11, maybe you need to set DISPLAY?");

-
-
     if (xkb_x11_setup_xkb_extension(conn,
                                     XKB_X11_MIN_MAJOR_XKB_VERSION,
                                     XKB_X11_MIN_MINOR_XKB_VERSION,
@@ -2057,6 +2055,17 @@ int main(int argc, char *argv[]) {
      * keyboard. */
     (void)load_keymap();

+    /* forcefully set keyboard layout group to 0 */
+    xcb_xkb_latch_lock_state(conn,
+                             XCB_XKB_ID_USE_CORE_KBD,
+                             0,
+                             0,
+                             true,
+                             0,
+                             0,
+                             0,
+                             0);
+
     /* Initialize the libev event loop. */
     main_loop = EV_DEFAULT;
     if (main_loop == NULL)
--
2.19.2
